test/Numerics/DGMethods/compressible_navier_stokes_equations/three_dimensional/ThreeDimensionalCompressibleNavierStokesEquations.jl

```diff
@@ -471,31 +431,74 @@ function numerical_flux_first_order!(
     return nothing
 end
 
-## Boundary Conditions
-
 boundary_conditions(model::CNSE3D) = model.boundary_conditions
 
 """
     boundary_state!(nf, ::CNSE3D, args...)
 applies boundary conditions for the hyperbolic fluxes
-dispatches to a function in OceanBoundaryConditions
+dispatches to a function in CNSEBoundaryConditions
 """
 @inline function boundary_state!(nf, bc, model::CNSE3D, args...)
-    return _ocean_boundary_state!(nf, bc, model, args...)
+    return _cnse_boundary_state!(nf, bc, model, args...)
 end
 
 """
-    ocean_boundary_state!(nf, bc::OceanBC, ::CNSE3D)
+    cnse_boundary_state!(nf, bc::FluidBC, ::CNSE3D)
 splits boundary condition application into velocity
 """
-@inline function ocean_boundary_state!(nf, bc::OceanBC, m::CNSE3D, args...)
-    ocean_boundary_state!(nf, bc.velocity, m, m.turbulence, args...)
-    ocean_boundary_state!(nf, bc.temperature, m, args...)
+@inline function cnse_boundary_state!(nf, bc::FluidBC, m::CNSE3D, args...)
+    cnse_boundary_state!(nf, bc.momentum, m, m.turbulence, args...)
+    cnse_boundary_state!(nf, bc.temperature, m, args...)
 
     return nothing
 end
 
 include("bc_momentum.jl")
-include("bc_tracer.jl")
+include("bc_temperature.jl")
+
+"""
+STUFF FOR ANDRE'S WRAPPERS
+"""
+
+function get_boundary_conditions(
+    model::SpatialModel{BL},
+) where {BL <: AbstractFluid3D}
+    bcs = model.boundary_conditions
+
+    west_east = (check_bc(bcs, :west), check_bc(bcs, :east))
+    south_north = (check_bc(bcs, :south), check_bc(bcs, :north))
+    bottom_top = (check_bc(bcs, :bottom), check_bc(bcs, :top))
+
+    return (west_east..., south_north..., bottom_top...)
+end
+
+function DGModel(model::SpatialModel{BL}) where {BL <: AbstractFluid3D}
+    params = model.parameters
+    physics = model.physics
+
+    Lˣ, Lʸ, Lᶻ = length(model.grid.domain)
+    bcs = get_boundary_conditions(model)
+    FT = eltype(model.grid.numerical.vgeo)
+    balance_law = CNSE3D{FT}(
+        (Lˣ, Lʸ, Lᶻ),
+        physics.advection,
+        physics.dissipation,
+        physics.coriolis,
+        physics.buoyancy,
+        bcs,
+        ρₒ = params.ρₒ,
+        cₛ = params.cₛ,
+    )
+
+    numerical_flux_first_order = model.numerics.flux # should be a function
+
+    rhs = DGModel(
+        balance_law,
```

Comment:
Why do we need to reconstruct a balance_law inside this DGModel constructor while the SpatialModel contains a balance_law?

Can we directly use `model.balance_law` here?

Reply:
Things could definitely be clearer. SpatialModel does not contain a balance law, but rather the tag for dispatch to construct a balance law

Reply:
Julia is weird that way